### PR TITLE
fix handling of mmcif with empty label_asym_id

### PR DESCRIPTION
**Pull request by dsehnal**

### CODE REVIEW

File: src/mol-io/reader/cif/schema.ts [hunk 1]
```diff
@@ -53,11 +53,18 @@ function getColumnCtor(t: Column.Schema): ColumnCtor {
...
+    }
+    return false;
+}
+
 function createColumn<T>(schema: Column.Schema, field: Data.CifField, value: (row: number) => T, toArray: Column<T>['toArray']): Column<T> {
     return {
         schema,
         __array: field.__array,
-        isDefined: field.isDefined,
+        isDefined: field.isDefined && hasPresentValues(field.rowCount, field.valueKind),
```
Comment (dsehnal) [inline src/mol-io/reader/cif/schema.ts:67]: I would prefer for this to be just part of the ``normalizeAtomSite`` function, as ``isDefined``'s sematics is essentially if it was present in the input file, not whether it has all values "undefined/unknown".

File: src/mol-model-formats/structure/basic/parser.ts [hunk 2]
```diff
@@ -150,17 +150,40 @@ function findModelEnd(num: Column<number>, startIndex: number) {
...
+    if (!table[b].isDefined || !hasPresentValues(table[b])) table[b] = table[a];
+}
+
+/** Fix possibly missing auth_/label_ columns */
+function getNormalizeAtomSite(atom_site: AtomSite) {
+    const normalized = Table.ofColumns(atom_site._schema, atom_site);
+    substUndefinedColumn(normalized, 'label_atom_id', 'auth_atom_id');
+    substUndefinedColumn(normalized, 'label_comp_id', 'auth_comp_id');
+    substUndefinedColumn(normalized, 'label_seq_id', 'auth_seq_id');
+    substUndefinedColumn(normalized, 'label_asym_id', 'auth_asym_id');
```
Comment (dsehnal) [inline src/mol-model-formats/structure/basic/parser.ts:171]: I wonder if this should also create a default ``entity_id`` column?

File: src/mol-model-formats/structure/basic/parser.ts [hunk 3]
```diff
@@ -150,17 +150,40 @@ function findModelEnd(num: Column<number>, startIndex: number) {
...
+    return normalized;
+}
+
 async function readStandard(ctx: RuntimeContext, data: BasicData, properties: CommonProperties, format: ModelFormat) {
     const models: Model[] = [];
 
     if (data.atom_site) {
-        const atomCount = data.atom_site.id.rowCount;
+        const normalizedAtomSite = getNormalizeAtomSite(data.atom_site);
```
Comment (dsehnal) [inline src/mol-model-formats/structure/basic/parser.ts:179]: Are the ``BasicData`` accessed somewhere else too?

**Review by dsehnal — COMMENTED** [review]
I am still a little paranoid, but this feels a lot better.

**Comment by arose** [comment]
@dsehnal one more try :)

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,8 @@ Note that since we don't clearly distinguish between a public and private interf
 
 ## [Unreleased]
 
+- Fix handling of mmcif with empty label_asym_id field
```
Comment (dsehnal) [inline CHANGELOG.md:9]:
```suggestion
- Fix handling of mmcif with empty ``label_*`` fields
```

**Review by dsehnal — COMMENTED** [review]
Ok, less paranoid now :) LGTM (can't approve as I created the PR).

**Pull request merged**